ext/node/polyfills/path/separator.ts

```diff
@@ -1,9 +1,11 @@
 // Copyright 2018-2025 the Deno authors. MIT license.
 
-// TODO(petamoriken): enable prefer-primordials for node polyfills
-// deno-lint-ignore-file prefer-primordials
-
 import { isWindows } from "ext:deno_node/_util/os.ts";
+import { primordials } from "ext:core/mod.js";
+
+const { SafeRegExp } = primordials;
 
 export const SEP = isWindows ? "\\" : "/";
-export const SEP_PATTERN = isWindows ? /[\\/]+/ : /\/+/;
+export const SEP_PATTERN = isWindows
+  ? new SafeRegExp("/[\\/]+/")
+  : new SafeRegExp("/\/+/");
```

Comment:
The leading and trailing slashes seem redundant?

<img width="298" alt="Screenshot 2025-04-07 at 16 35 49" src="https://github.com/user-attachments/assets/b5985120-c34e-4ab3-a2a5-90342edef3e2" />

Also the backslashes in the middle need to be doubled?

Reply:
Good catch! Fixed.